### PR TITLE
Cleanup and performance improvements

### PR DESCRIPTION
**Pull request by schiessle**
* some performance improvements, only compute one set of data per run
* move API to a OCSController

### CODE REVIEW

File: appinfo/routes.php [hunk 1]
```diff
@@ -35,5 +35,8 @@
 		// api
 		['name' => 'api#add', 'url' => '/api/v1/data', 'verb' => 'POST'],
 		['name' => 'api#get', 'url' => '/api/v1/data', 'verb' => 'GET'],
-    ]
+    ],
```
Comment (nickvergessen) [inline appinfo/routes.php]: tab?

**Review by nickvergessen — APPROVED** [review]

File: appinfo/routes.php (hunk 1 above)
Comment (schiessle) [reply appinfo/routes.php]: fixed

File: appinfo/info.xml [hunk 2]
```diff
@@ -9,6 +9,6 @@
     <namespace>Survey_Server</namespace>
     <category>other</category>
     <dependencies>
-        <nextcloud min-version="9" max-version="13"/>
+        <nextcloud min-version="9" max-version="14"/>
```
Comment (nickvergessen) [inline appinfo/info.xml:12]: Forgot to submit my review.

I'd just lift the min. to the last supported version: 13

Comment (MorrisJobke) [reply appinfo/info.xml:12]: :+1:

**Pull request merged**